### PR TITLE
DIOS-3792 Fix AV sync of shaka recordings

### PR DESCRIPTION
**Pull request by bcostdolby**

### CODE REVIEW

**Review by harryz2000 — APPROVED** [review]
lgtm

**Comment by bcostdolby** [comment]
@mildsunrise Please approve if happy with Harrys review (doing two reviewer process now)

**Comment by harryz2000** [comment]
My understanding of the idea of two stage review is to ensure previous comments made by reviewer were fixed/responded and add new comments if needed. The approval shouldn't depend on whether approver is happy with the reviewer's comment but whether he/she is happy with the code. If there is a disagreement with the comments, all the code author, reviewer and approver can work together to find a solution. I guess if two of three agree on a solution, it then can go that way. It would be still nice everyone is on the same page though.

File: src/MediaFrameListenerBridge.cpp [hunk 1]
```diff
@@ -271,6 +274,21 @@ void MediaFrameListenerBridge::onMediaFrame(DWORD ignored, const MediaFrame& fra
...
+				//is 0. In this case we will estimate it based on the target fps which the transcoder
+				//does set which is sufficient for this use case.
+				if (frame->GetDuration() == 0 && video->GetTargetFps() != 0)
+				{
+					pendingDuration = 1000.0 / video->GetTargetFps();
+				}
+				else
+				{
+					pendingDuration = frame->GetDuration() * 1000 / rate;
+				}
```
Comment (murillo128) [inline src/MediaFrameListenerBridge.cpp:291]: how about?

```
				if (frame->GetDuration())
				{
					pendingDuration = 1000.0 / video->GetTargetFps();
				}
				else if (video->GetTargetFps())
				{
					pendingDuration = frame->GetDuration() * 1000 / rate;
				} else {
					pendingDuration = 8; //Use a default of 10ms for smooth sending (max 125fps).
   				}
```

**Review by murillo128 — APPROVED** [review]
lgtm

File: src/MediaFrameListenerBridge.cpp (hunk 1 above)
Comment (bcostdolby) [reply src/MediaFrameListenerBridge.cpp:291]: Yes I like this. It is just a bit clearer and will add it in once I get the CI for the tests going properly.

**Pull request merged**